src/effects/specialeffectchains.cpp

```diff
@@ -0,0 +1,232 @@
+        const int iEffectSlotNumber) {
+    return QString("[EffectRack1_EffectUnit%1_Effect%2]")
+            .arg(QString::number(iChainSlotNumber + 1),
+                    QString::number(iEffectSlotNumber + 1));
+}
+
+OutputEffectChain::OutputEffectChain(EffectsManager* pEffectsManager,
+        EffectsMessengerPointer pEffectsMessenger)
+        : EffectChain(formatEffectChainGroup("[Master]"),
+                  pEffectsManager,
+                  pEffectsMessenger,
+                  SignalProcessingStage::Postfader) {
+    addEffectSlot("[OutputEffectRack_[Master]_Effect1]");
+    m_effectSlots[0]->setEnabled(true);
+
+    // Register the master channel
+    const ChannelHandleAndGroup* masterHandleAndGroup = nullptr;
+
+    // TODO(Be): Remove this hideous hack to get the ChannelHandleAndGroup
+    const QSet<ChannelHandleAndGroup>& registeredChannels =
+            m_pEffectsManager->registeredInputChannels();
+    for (const ChannelHandleAndGroup& handle_group : registeredChannels) {
+        if (handle_group.name() == "[MasterOutput]") {
+            masterHandleAndGroup = &handle_group;
+            break;
+        }
+    }
+    DEBUG_ASSERT(masterHandleAndGroup != nullptr);
+
+    registerInputChannel(*masterHandleAndGroup);
+    enableForInputChannel(*masterHandleAndGroup);
+    m_pControlChainMix->set(1.0);
+    sendParameterUpdate();
+}
+
+QString OutputEffectChain::formatEffectChainGroup(
+        const QString& group) {
+    return QString("[OutputEffectRack_%1]").arg(group);
+}
+
+PerGroupEffectChain::PerGroupEffectChain(const QString& group,
+        const QString& chainSlotGroup,
+        SignalProcessingStage stage,
+        EffectsManager* pEffectsManager,
+        EffectsMessengerPointer pEffectsMessenger)
+        : EffectChain(chainSlotGroup,
+                  pEffectsManager,
+                  pEffectsMessenger,
+                  stage) {
+    // Set the chain to be fully wet.
+    m_pControlChainMix->set(1.0);
+    sendParameterUpdate();
+
+    // TODO(rryan): remove.
+    const ChannelHandleAndGroup* handleAndGroup = nullptr;
+    for (const ChannelHandleAndGroup& handle_group :
+            m_pEffectsManager->registeredInputChannels()) {
+        if (handle_group.name() == group) {
+            handleAndGroup = &handle_group;
+            break;
+        }
+    }
+    DEBUG_ASSERT(handleAndGroup != nullptr);
+
+    // Register this channel alone with the chain slot.
+    registerInputChannel(*handleAndGroup);
+    enableForInputChannel(*handleAndGroup);
+}
+
+QuickEffectChain::QuickEffectChain(const QString& group,
+        EffectsManager* pEffectsManager,
+        EffectsMessengerPointer pEffectsMessenger)
+        : PerGroupEffectChain(group,
+                  formatEffectChainGroup(group),
+                  SignalProcessingStage::Postfader,
+                  pEffectsManager,
+                  pEffectsMessenger) {
+    for (int i = 0; i < kNumEffectsPerUnit; ++i) {
```

Comment:
No, the same constant needs to be used for both StandardEffectChain and QuickEffectChain, otherwise chain presets might not work in QuickEffectChains.